File: src/main/java/io/zeitwert/fm/task/model/DocTask.java

package io.zeitwert.fm.task.model;

import java.time.OffsetDateTime;

import io.zeitwert.ddd.aggregate.model.Aggregate;
import io.zeitwert.ddd.doc.model.Doc;
import io.zeitwert.fm.account.model.ItemWithAccount;
import io.zeitwert.fm.account.model.ObjAccount;
import io.zeitwert.fm.collaboration.model.ItemWithNotes;
import io.zeitwert.fm.task.model.enums.CodeTaskPriority;

public interface DocTask extends Doc, ItemWithAccount, ItemWithNotes {

	Integer getRelatedToId();

	void setRelatedToId(Integer id);

	Aggregate getRelatedTo();

	@Override
	Integer getAccountId();

	@Override
	void setAccountId(Integer id);

	@Override
	ObjAccount getAccount();

	String getSubject();

	void setSubject(String subject);

	String getContent();

	void setContent(String content);

	Boolean getIsPrivate();

	void setIsPrivate(Boolean isPrivate);

	CodeTaskPriority getPriority();

	void setPriority(CodeTaskPriority priority);

	OffsetDateTime getDueAt();

	void setDueAt(OffsetDateTime dueAt);

	OffsetDateTime getRemindAt();

	void setRemindAt(OffsetDateTime remindAt);

}

File: fm-dddrive/src/test/java/io/dddrive/domain/task/model/DocTask.kt
package io.dddrive.domain.task.model

import io.dddrive.doc.model.Doc
import io.dddrive.domain.task.model.enums.CodeTaskPriority
import io.dddrive.property.model.PartListProperty
import java.time.OffsetDateTime

interface DocTask : Doc {

	var subject: String?
	var content: String?
	var isPrivate: Boolean?
	var priority: CodeTaskPriority?
	var dueAt: OffsetDateTime?
	var remindAt: OffsetDateTime?

	val commentList: PartListProperty<DocTaskPartComment>

}
